lib/rubocop/cop/layout/space_inside_array_literal_brackets.rb

```diff
@@ -0,0 +1,241 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Layout
+      # Checks that brackets used for array literals have or don't have
+      # surrounding space depending on configuration.
+      #
+      # @example EnforcedStyle: space
+      #   # The `space` style enforces that array literals have
+      #   # surrounding space.
+      #
+      #   # bad
+      #   array = [a, b, c, d]
+      #
+      #   # good
+      #   array = [ a, b, c, d ]
+      #
+      # @example EnforcedStyle: no_space
+      #   # The `no_space` style enforces that array literals have
+      #   # no surrounding space.
+      #
+      #   # bad
+      #   array = [ a, b, c, d ]
+      #
+      #   # good
+      #   array = [a, b, c, d]
+      #
+      # @example EnforcedStyle: compact
+      #   # The `compact` style normally requires a space inside
+      #   # array brackets, with the exception that successive left
+      #   # or right brackets are collapsed together in nested arrays.
+      #
+      #   # bad
+      #   array = [ a, [ b, c ] ]
+      #
+      #   # good
+      #   array = [ a, [ b, c ]]
+      class SpaceInsideArrayLiteralBrackets < Cop
+        include SurroundingSpace
+        include ConfigurableEnforcedStyle
+
+        MSG = '%<command>s space inside array brackets.'.freeze
+        EMPTY_MSG = '%<command>s space inside empty array brackets.'.freeze
+
+        def on_array(node)
+          left, right = array_brackets(node)
+          return if right.nil?
```

Comment:
This guard clause is needed for 3 of the `cli_autocorrect` tests that seem to trigger `on_array` with incomplete arrays.

Reply:
I guess those tests should be fixed instead.

Reply:
These are the 3 failing test without that check. Notice that 2 of them do not even contain square brackets.

```
rspec ./spec/rubocop/cli/cli_autocorrect_spec.rb:1017 # RuboCop::CLI can correct WordArray and SpaceAfterComma offenses
rspec ./spec/rubocop/cli/cli_autocorrect_spec.rb:312 # RuboCop::CLI corrects IndentationWidth, RedundantBegin, and RescueEnsureAlignment offenses
rspec ./spec/rubocop/cli/cli_autocorrect_spec.rb:411 # RuboCop::CLI corrects InitialIndentation offenses
```

I've added a commit with more detail further down the PR.